src/web3/get/queryRewardsRetrievedEvents.ts

```diff
@@ -56,6 +59,53 @@ export const queryRewardsRetrieved = async (
 
       return datum;
     });
+
+    // On mainnet, look back at old voting contracts to retrieve rewards
+    // for voters who have voted on previous iteration of the contract.
+    // this is largely just for summation of Total UMA Collected.
+    if (
+      process.env.REACT_APP_CURRENT_ENV === "main" ||
+      process.env.REACT_APP_CURRENT_ENV === undefined
+    ) {
+      const promises = OLD_VOTING_CONTRACT_ADDRESSES.map(async (addr) => {
```

Comment:
can we generalize this to a function which takes any voting contract address? Or at least a function which handles current voting contract and previous ones.

`async getRewards(userAddress,contractAddress)`

and then do something like this
```js
const rewards = await Promise.all(
  allVotingContractAddresses.map(contractAddress=>getRewards(userAddress,contractAddress,provider))
)
return rewards.flat()
```

wdyt?

Reply:
I think the issue is the ABI is different between the newest contract and oldest. That's why I'm forced to create a new version of the old contract.

i could add the ABI in an arg, but it gets messier somewhat quickly. Needing to reference the old contract(s) is awkward.